File: v3-to-do-list-api-fetch-tailwind/frontend/src/services/tasks.js
const API_BASE_URL = process.env.API_URL || "http://localhost:8000"


export async function getTasks(){
    try {
        const response = await fetch(`${API_BASE_URL}/api/tasks/`)
        const data = await response.json()
        const tasks = data.tasks
        return true, tasks
    } catch (error) {
        console.log(error)
        return false, "request error"
    }
}

export async function getTaskById(taskId){
    try {
        const response = await fetch(`${API_BASE_URL}/api/tasks/${taskId}/`)
        const task = await response.json()
        return true, task
    } catch (error) {
        console.log(error)
        return false, "request error"
    }
}

export async function createTask(taskObj){
    try {
        const response = await fetch(`${API_BASE_URL}/api/tasks/`,{
            method: "POST",
            body: JSON.stringify(taskObj),
            headers: {
              "Content-Type": "application/json",
            },
        })
        const data = await response.json()
        return true, data
    } catch (error) {
        console.log(error)
        return false, "request error"
    }
}

export async function updateTaskById(taskId, taskObj){
    try {
        const response = await fetch(`${API_BASE_URL}/api/tasks/${taskId}/`,{
            method: "PUT",
            body: JSON.stringify(taskObj),
            headers: {
              "Content-Type": "application/json",
            },
        })
        const data = await response.json()
        return true, data
    } catch (error) {
        console.log(error)
        return false, "request error"
    }
}

export async function deleteTaskById(taskId){
    try {
        const response = await fetch(`${API_BASE_URL}/api/tasks/${taskId}/`,{
            method: "DELETE"
        })
        const data = await response.json()
        return true, data
    } catch (error) {
        console.log(error)
        return false, "request error"
    }
}

File: v6-to-do-list-go-inmemory-react/backend/app/services/tasks.go
package services

import (
	"to_do_list/app/repository"
	"to_do_list/app/schemas"
)

func GetTasks() ([]schemas.TaskSchema, error) {
	tasks, err := repository.GetTasks()
	if err != nil {
		return nil, err
	}
	return tasks, nil
}

func GetTaskById(taskId int) (schemas.TaskSchema, error) {

	task, err := repository.GetTaskById(taskId)
	if err != nil {
		return schemas.TaskSchema{}, err
	}
	return task, nil
}

func CreateTask(taskData schemas.TaskSchema) (schemas.TaskSchema, error) {
	task, err := repository.CreateTask(taskData)
	if err != nil {
		return schemas.TaskSchema{}, err
	}
	return task, nil
}

func UpdateTaskById(taskId int, taskData schemas.UpdateTaskSchema) (schemas.TaskSchema, error) {
	task, err := repository.UpdateTaskById(taskId, taskData)
	if err != nil {
		return schemas.TaskSchema{}, err
	}
	return task, nil
}

func DeleteTaskById(taskId int) error {
	err := repository.DeleteTaskById(taskId)
	if err != nil {
		return err
	}
	return nil
}
